chromium_src/crypto/symmetric_key.cc

```diff
@@ -15,25 +15,26 @@ namespace crypto {
 // static
 std::unique_ptr<SymmetricKey>
 SymmetricKey::DeriveKeyFromPasswordUsingPbkdf2Sha256(
-    Algorithm algorithm,
+    Algorithm,
     const std::string& password,
     const std::string& salt,
     size_t iterations,
     size_t key_size_in_bits) {
-  if (!CheckDerivationParameters(algorithm, key_size_in_bits))
+  // Only doing checks for AES keys for now, as `SymmetricKey` will be deleted
```

Comment:
what are we using this for in Brave?

Reply:
This got reworked a bit in a later commit (https://github.com/brave/brave-core/pull/25662/commits/cc2e2884afad18ecc60a7a9c4260742b30f9d758), but it's used in components/brave_wallet/browser/internal/hd_key.cc (HDKey::GenerateFromV3UTC).

Reply:
Basically we don't want to crash the browser when for example importing from MetaMask with bad data